controller/setting_controller.go

```diff
@@ -351,6 +373,54 @@ func (sc *SettingController) updateTaintToleration() error {
 	return nil
 }
 
+func updateTolerationForDeployment(ds *appsv1.Deployment, lastAppliedTolerations, newTolerations []v1.Toleration) error {
```

Comment:
Is it possible to use a common function rather than 2 separate functions here? Notice that both functions will update `PodTemplateSpec` and `Annotations`

Reply:
It would be something like this:
```
func (sc *SettingController) updateTolerationAndAnnotation(podTemplateSpec *v1.PodTemplateSpec, obj runtime.Object, lastAppliedTolerations, newTolerations []v1.Toleration) error {
	existingTolerationsMap := util.TolerationListToMap(podTemplateSpec.Spec.Tolerations)
	lastAppliedTolerationsMap := util.TolerationListToMap(lastAppliedTolerations)
	newTolerationsMap := util.TolerationListToMap(newTolerations)
	podTemplateSpec.Spec.Tolerations = getFinalTolerations(existingTolerationsMap, lastAppliedTolerationsMap, newTolerationsMap)
	newTolerationsByte, err := json.Marshal(newTolerations)
	if err != nil {
		return err
	}
	if err := util.SetAnnotation(obj, types.GetLonghornLabelKey(types.LastAppliedTolerationAnnotationKeySuffix), string(newTolerationsByte)); err != nil {
		return err
	}
	return nil
}
///////////////////////////////
sc.updateTolerationAndAnnotation(dp.Spec.Template, dp, lastAppliedTolerationsList, newTolerationsList)
//////////////////////////////
sc.updateTolerationAndAnnotation(ds.Spec.Template, ds, lastAppliedTolerationsList, newTolerationsList)
```

imho, the separation of the 2 functions is a bit repeating code but it is easier to read.  
However, if you prefer the compact form, it totally makes sense too. 
Please let me know which way you prefer?

Reply:
OK. Let's retain the current implementation